addons/udes_stock/wizard/change_quant_location.py

```diff
@@ -43,6 +88,24 @@ def create_picking(self):
         if self.reference:
             params['origin'] = self.reference
 
+        # If we allow users to create these pickings for reserved quants, delete
+        # moves/movelines on other pickings.
+        # If we don't allow it the validation in create_picking will error
+        if self._check_reserved_packages_permitted():
+            # Remove moves/move lines for quants in other pickings
+            movelines = packages.get_move_lines()
+            # First split out the movelines from the original pickings
+            # to preserve the original picking and any moves that are for quants
+            # in different pacakges.
+            new_moves = Move.browse()
+            for move, movelines in movelines.groupby(lambda ml: ml.move_id):
```

Comment:
Have we considered using backorder move lines so we don't cancel and re-reserve the stock ? This will imply that we have to remove quant_ids from params and create the picking before.

Reply:
I did consider and try that. The problem then was that we were adding quants at multiple points which caused issues with various validations. I did it this way so that all quants were added at once to avoid those issues

Reply:
Is this when is not all the package reserved? I would restrict this to be only used in full reserved packages.

Reply:
Or if we were trying to move a pallet when some packages were reserved and some weren't. I don't know if restricting it to just fully reserved packages is drawing an arbitrary line